src/hooks/useSubmitButtonVisibility.ts

```diff
@@ -0,0 +1,72 @@
+import {useEffect, useRef, useState} from 'react';
+import {Dimensions} from 'react-native';
+import getPlatform from '@libs/getPlatform';
+import CONST from '@src/CONST';
+import useSafeAreaInsets from './useSafeAreaInsets';
+import useThemeStyles from './useThemeStyles';
+import useWindowDimensions from './useWindowDimensions';
+
+// Useful when there's a need to hide the submit button from FormProvider,
+// to let form content fill the page when virtual keyboard is shown
+function useSubmitButtonVisibility() {
+    const styles = useThemeStyles();
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const [isSubmitButtonVisible, setIsSubmitButtonVisible] = useState(true);
+    const initialWindowHeightRef = useRef(windowHeight);
+    const isSmallScreenWidthRef = useRef(isSmallScreenWidth);
+    const {bottom} = useSafeAreaInsets();
+    const platform = getPlatform();
+    const isNative = platform === CONST.PLATFORM.IOS || platform === CONST.PLATFORM.ANDROID;
```

Comment:
Can we somehow do this with platform specific files? That's what our guidelines are.

https://github.com/Expensify/App/blob/db01cf5b4aa52069cf416d7a1a84ea3bdbfeea8f/README.md?plain=1#L480

Reply:
I'm pretty sure we can do it, since I already see some hooks with platforms specific files.